### PR TITLE
Feat/notebookendpoints

### PR DESCRIPTION
**Pull request by aasimkhan30**
Added delete notebook endpoints and made some changes queries to handle space sensitive databases.

### CODE REVIEW

**Comment by Charles-Gagnon** [comment]
>     /// SQL Agent Notebook templates params 

Looks like these doc comments should be updated - most of the ones below here seem to be inaccurate

---
Refers to: src/Microsoft.SqlTools.ServiceLayer/Agent/Contracts/AgentNotebookRequest.cs:217 in 77f9b0f. [](commit_id = 77f9b0f43222d3c94d0f406b30adcd30a53b9fd0, deletion_comment = False)

File: src/Microsoft.SqlTools.ServiceLayer/Agent/Contracts/AgentNotebookRequest.cs [hunk 1]
```diff
@@ -256,4 +256,32 @@ public static readonly
...
+        public string TargetDatabase { get; set; }
+        public AgentNotebookHistoryInfo agentNotebookHistory { get; set; }
+    }
+
+    /// <summary>
+    /// SQL Agent Notebook materialized result
+    /// </summary>
+    public class DeleteNotebookMaterializedResult : ResultStatus
+    {
+        public string NotebookMaterialized { get; set; }
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/Agent/Contracts/AgentNotebookRequest.cs]: >NotebookMaterialized [](start = 22, length = 20)

Is this supposed to be the name of the notebook that was deleted? Suggest having the name be MaterializedNotebook then - right now this sounds like it's telling us whether the notebook was materialized (which makes no sense in this context)

File: src/Microsoft.SqlTools.ServiceLayer/Agent/Contracts/AgentNotebookRequest.cs [hunk 2]
```diff
@@ -256,4 +256,32 @@ public static readonly
...
+    }
+
+    /// <summary>
+    /// SQL Agent Notebook materialized request type
+    /// </summary>
+    public class DeleteNotebookMaterializedRequest
+    {
+        public static readonly
+            RequestType<DeleteNotebookMaterializedParams, ResultStatus> Type =
+            RequestType<DeleteNotebookMaterializedParams, ResultStatus>.Create("agent/deletenotebookmaterialized");
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/Agent/Contracts/AgentNotebookRequest.cs]: >deletenotebookmaterialized [](start = 86, length = 26)

Same here - this seems like it should probably be agent/deletematerializednotebook?

File: src/Microsoft.SqlTools.ServiceLayer/Agent/Jobs/AgentNotebookHelper.cs [hunk 3]
```diff
@@ -200,7 +200,7 @@ INNER JOIN
                 notebook.JobId,
                 notebook.TargetDatabase);
 
-           
+
         }
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/Agent/Jobs/AgentNotebookHelper.cs]: nit: get rid of extra lines here

File: src/Microsoft.SqlTools.ServiceLayer/Agent/Jobs/AgentNotebookHelper.cs [hunk 4]
```diff
@@ -567,79 +556,143 @@ UPDATE notebooks.nb_template
...
                 updateMaterializedNotebookPinQuery,
                 updateMaterializedNotebookPinParams,
                 targetDatabase);
         }
+
+        /// <summary>
+        /// Delete a particular run of the job. Deletion mainly including clearing out the notebook,
+        /// and notebook-error. The API doesn't delete the row because some notebook runs that have job
+        /// error in them don't have an entry in the materialized table. For keeping track of those notebook
+        /// runs the entry is added into the table with is_delete set to 1.
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/Agent/Jobs/AgentNotebookHelper.cs]: Why do we want to keep track of these? If we're deleting them isn't that implying we don't care about them anymore?

Comment (aasimkhan30) [reply src/Microsoft.SqlTools.ServiceLayer/Agent/Jobs/AgentNotebookHelper.cs]: The API only deletes the artifacts created by notebook job history and not the history of the notebook job. So if the job has an error there won't be any entry in the materialized table. Therefore, just to remove the job from the UI I'm setting the 'is_deleted to false.

File: src/Microsoft.SqlTools.ServiceLayer/Agent/AgentService.cs [hunk 5]
```diff
@@ -1470,7 +1470,36 @@ internal async Task HandleUpdateAgentNotebookRunPinRequest(UpdateAgentNotebookRu
...
+                        connInfo,
+                        parameters.agentNotebookHistory,
+                        parameters.TargetDatabase);
+                    result.Success = true;
+                }
+                catch (Exception e)
+                {
+                    result.Success = false;
+                    result.ErrorMessage = e.ToString();
+
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.ServiceLayer/Agent/AgentService.cs]: Normally you'd just do await result.SendError(e);

Comment (Charles-Gagnon) [reply src/Microsoft.SqlTools.ServiceLayer/Agent/AgentService.cs]: Make an issue to make these exceptions call SendError instead

---
In reply to: [321924046](https://github.com/microsoft/sqltoolsservice/pull/862#discussion_r321924046) [](ancestors = 321924046)

**Pull request merged**